docs/ardana/testing.md

```diff
@@ -210,6 +210,229 @@ prepare_infra
 ...
 ```
 
+## Custom Ardana Gerrit CI builds
+
+The default integration test job launched by the Gerrit Jenkins CI uses
+a `standard` generated input model, with a standalone deployer, 2
+controller nodes and one SLES compute node. It also runs against a
+`develcloud` cloud media, on top of which it applies the most recent
+Ardana ansible changes and, of course, the target Gerrit change under
+test and all its Gerrit change dependencies.
+
+This is sufficient for most cases, but sometimes there are special
+circumstances that require one or more additional, customized
+`openstack-ardana` type integration jobs to be executed against the same
+Gerrit change and to have them report their results back to Gerrit,
+where they can be tracked and verified by all stake holders.
+
+This is now supported by allowing users to trigger manual builds for
+the parameterized Jenkins job that runs the Gerrit CI for either
+[cloud 8](https://ci.nue.suse.com/job/openstack-ardana-gerrit-cloud8/build)
+or [cloud 9](https://ci.nue.suse.com/job/openstack-ardana-gerrit-cloud9/build),
+and supplying custom values for its parameters, to achieve different
+results:
+
+* `GERRIT_CHANGE_NUMBER` - this parameter must be set to the target Gerrit
+change number value
+
+* `voting` - this flag controls whether the Gerrit job posts `Verify+2`
+or `Verify-2` label values on the Gerrit change. Non-voting jobs also
+post progress and outcome messages, but do not alter the `Verify` label
+value. There can be at most one voting job running for a Gerrit change
+at any given time. The most recent voting job automatically cancels
+older voting jobs running for the same Gerrit change.
+
+* `gerrit_context` - this is a string value that uniquely identifies
+a job build within the set of builds running against the same Gerrit
+change. A unique `gerrit_context` parameter value should be used for
+each additional job that is manually triggered against a Gerrit
+change, otherwise it will supersede and automatically cancel the
+previous jobs running with the same `gerrit_context` value on the same
+Gerrit change.
+This parameter should be set to a short string describing what the job
+actually achieves, as it is included both in the Gerrit build status
+reports and the Jenkins build description.
+
+* `integration_test_job` - the name of an existing Ardana Jenkins job
+that is triggered to run integration tests on the target Gerrit change.
+This parameter may be used to override the default
+`cloud-ardana8-job-std-min-gerrit-x86_64` and `cloud-ardana8-job-std-min-gerrit-x86_64`
+jobs currently employed by the Ardana CI to run integration tests.
+If the custom integration job also requires its parameters to be overridden,
+this can be achieved via the `extra_params` parameter.
+If this parameter is reset to an empty value, no integration tests will
```

Comment:
Do we have a sanity test that fails if this parameter is empty and you have voting mode enabled? I'd prefer that we have a special commit check only job or something that can be used as a minimal job if we want to force a Verified status update... I'm not sure I can see a reason for a voting job with no integration test...

Reply:
I removed that use-case completely.